pkg/controllers/volume-mgmt/handler.go

```diff
@@ -757,3 +757,32 @@ func (c *CStorVolumeController) reconcileVersion(cv *apis.CStorVolume) (*apis.CS
 	}
 	return cv, nil
 }
+
+// markCVStatusToOffline will fetch all the CV resources present
+// in etcd and mark it them CV.Status to Offline
+func (c *CStorVolumeController) markCVStatusToOffline() {
+	openebsNameSpace := os.Getenv("OPENEBS_NAMESPACE")
+	cvList, err := c.clientset.CstorV1().CStorVolumes(openebsNameSpace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("failed to fetch CV list, error: %v", err)
+		return
+	}
+	for i := range cvList.Items {
+		// Checking if Valid Cstor
+		cv := cvList.Items[i]
+		if IsValidCStorVolumeMgmt(&cv) {
+			cv.Status.Phase = apis.CVStatusOffline
```

Comment:
Let's add check only if phase is non-empty(to maintain in sync with cvr & cspc status updation
```suggestion
                        if cv.Status.Phase == "" {
                              return
                        }
			cv.Status.Phase = apis.CVStatusOffline
```

Reply:
I think you meant, If the phase is empty then continue. instead return right? [cvr commit](https://github.com/openebs/cstor-operators/pull/349/commits/d82e6ddfe20bda7f5005a03d2066fe928cd385c7#diff-214b27e3b1e501677482b660967b0f500018e110cae11d3f2505642d27fba096R723)

Reply:
Also haven't test the changes yet. I have to setup and do that as well. Any documentation I can look at to test my changes?

Reply:
> I think you meant, If the phase is empty then continue. instead return right? 

No In CV case we can `return` since there is only a one-one mapping between cStor target pod and CStorVolume, if execution is happening here then we already found the right CV so not required to continue on others.

Reply:
> Also haven't test the changes yet. I have to setup and do that as well. Any documentation I can look at to test my changes?

Well!! 
- Please go through [prerequisites](https://github.com/openebs/cstor-operators/blob/develop/docs/developer-guide/devel.md#prerequisites) for the dev environment( Note: You can have the latest Go version also).
- Build the cStor-volume image using `make volume-manager-image` for building all images you can click [here](https://github.com/openebs/cstor-operators/blob/develop/docs/developer-guide/devel.md#build-the-images) 
- Once the image is built successfully, launch [minikube](https://minikube.sigs.k8s.io/docs/start/) cluster and follow [quick_start](https://github.com/openebs/cstor-operators/blob/develop/docs/quick.md#quickstart) doc to provision volume.
  Note: Minikube cluster might not have external disks, you can enable [sparse file count](https://github.com/openebs/charts/blob/ab0e45c96cd73f2f45c4d5431d3e7fa0ec903642/cstor-operator.yaml#L5317) to create sparse based blockdevices and use them for creating pools.

Once the volume is provisioned successfully, Scale down the cStor target pod, pod name starts with `pvc-XXX-YYYY....` and you should see corresponding resource i.e CV should mark as `Offline`

Reply:
You can also report if there is any issue related to dev setup in [#openebs-dev](https://kubernetes.slack.com/archives/C015124PVTL) channel.